### PR TITLE
 fix ssm jobs

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description: What code changed, and why?
we now have to add a token to an ingest policy after creation for that token to be a valid ingest token; this change makes that happen

### :chains: Related Resources
https://github.com/chef/automate/issues/3394

### :+1: Definition of Done
tokens created for ssm jobs work

### :athletic_shoe: How to Build and Test the Change
to truly test this, you need to bring up the dev env in aws and create an ec2 node integration, then try scanning some ssm nodes

the other way to test locally is to call the runssmjob function from the regular scan job path, log the token, and then ensure the token works for sending in a report and fetching a profile

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/compliance-service/compliance.go [hunk 1]
```diff
@@ -554,22 +554,36 @@ func setup(ctx context.Context, connFactory *secureconn.Factory, conf config.Com
...
-		authClient := auth.NewTokensMgmtClient(authConn)
-		if authClient == nil {
+		authnClient := authn.NewTokensMgmtClient(authnConn)
+		if authnClient == nil {
 			logrus.Errorf("serveGrpc got nil for NewTokensMgmtClient: %s", err)
 			return err
 		}
+		// get authz connection
+		authzConn, err = connFactory.DialContext(timeoutCtx, "authz-service", fmt.Sprintf("%s:%d", conf.Authz.HostBind, conf.Authz.Port),
+			grpc.WithBlock())
```
Comment (vjeffrey) [inline components/compliance-service/compliance.go:571]: we already have the authz host/port info bc we use it during project tagging and updating

File: components/compliance-service/inspec-agent/remote/remote.go [hunk 2]
```diff
@@ -29,16 +33,28 @@ func RunSSMJob(ctx context.Context, ssmJob *types.InspecJob) *inspec.Error {
 	logrus.Debugf("Running ssm job: %+v", ssmJob)
 	// 1. CREATE TOKEN
 	// this is needed to hand over to inspec so it can report to automate
-	if RemoteJobInfo.TokensMgmtClient == nil {
+	if RemoteJobInfo.TokensMgmtClient == nil || RemoteJobInfo.PoliciesClient == nil {
```
Comment (vjeffrey) [inline components/compliance-service/inspec-agent/remote/remote.go:36]: make sure we fail if we dont have the policies client

File: components/compliance-service/inspec-agent/remote/remote.go [hunk 3]
```diff
@@ -29,16 +33,28 @@ func RunSSMJob(ctx context.Context, ssmJob *types.InspecJob) *inspec.Error {
...
+		Name:   tokenID,
 		Active: true,
 	})
 	if err != nil {
-		logrus.Errorf("unable to create auth token for reporting to automate: %s", err.Error())
+		logrus.Errorf("unable to create auth token for reporting to automate; aborting job %s: %s", ssmJob.JobName, err.Error())
+		return translateToInspecErr(err)
+	}
+	// add token to ingest policy
+	_, err = RemoteJobInfo.PoliciesClient.AddPolicyMembers(ctx, &v2.AddPolicyMembersReq{
```
Comment (vjeffrey) [inline components/compliance-service/inspec-agent/remote/remote.go:52]: this is the real new stuff/the fix -- adding the token to the ingest policy

File: components/compliance-service/inspec-agent/remote/remote.go [hunk 4]
```diff
@@ -29,16 +33,28 @@ func RunSSMJob(ctx context.Context, ssmJob *types.InspecJob) *inspec.Error {
 	logrus.Debugf("Running ssm job: %+v", ssmJob)
 	// 1. CREATE TOKEN
 	// this is needed to hand over to inspec so it can report to automate
-	if RemoteJobInfo.TokensMgmtClient == nil {
+	if RemoteJobInfo.TokensMgmtClient == nil || RemoteJobInfo.PoliciesClient == nil {
 		logrus.Error("unable to create auth token")
 		return translateToInspecErr(fmt.Errorf("unable to connect to auth client: aborting job run for job %+v", ssmJob))
 	}
+	tokenID := fmt.Sprintf("inspec-to-automate-scanjob-%s", ssmJob.JobID)
+	ctx = auth_context.NewOutgoingContext(auth_context.NewContext(ctx, []string{"tls:service:deployment-service:internal"}, []string{}, "", ""))
```
Comment (msorens) [inline components/compliance-service/inspec-agent/remote/remote.go:41]: You are inadvertently masquerading as the deployment service here. That may be OK but there is a check for that in the gateway that is used for a specific purpose: https://github.com/chef/automate/blob/c8f110926945f0363759effd949e3cab1218fd62/components/automate-gateway/gateway/middleware/auth_interceptor.go#L246-L250

Probably better to use your own designator, perhaps:
```suggestion
	ctx = auth_context.NewOutgoingContext(auth_context.NewContext(ctx, []string{"tls:service:compliance-service:internal"}, []string{}, "", ""))
```

Comment (vjeffrey) [reply components/compliance-service/inspec-agent/remote/remote.go:41]: ah yes! thank you!

**Pull request merged**